core/src/message/context.rs

```diff
@@ -313,6 +313,43 @@ impl MessageContext {
         }
     }
 
+    /// Pushes the incoming buffer/payload into stored payload by handle.
+    pub fn send_push_input(&mut self, handle: u32, range: CheckedRange) -> Result<(), Error> {
+        let data = self
+            .store
+            .outgoing
+            .get_mut(&handle)
+            .ok_or(Error::OutOfBounds)?
+            .as_mut()
+            .ok_or(Error::LateAccess)?;
+
+        let CheckedRange {
+            offset,
+            excluded_end,
+        } = range;
+
+        data.try_extend_from_slice(&self.current.payload()[offset..excluded_end])
+            .map_err(|_| Error::MaxMessageSizeExceed)?;
+
+        Ok(())
+    }
+
+    /// Check range for `send_push_input`/`reply_push_input` methods.
```

Comment:
Can you explain in doc what is this check about?

So far I don't see this functions checks anything but rather returns some different range

Reply:
ok